### PR TITLE
Add iterators on ASTs, allowing for loops, std::find/std::find_if, std::count/std::count_if, etc.

### PR DESCRIPTION
**Pull request by flomnes**

### CODE REVIEW

File: src/solver/expressions/iterators/pre-order.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    Node* current = nodeStack.top();
+    nodeStack.pop();
+
+    const auto children = childrenLeftToRight(current);
+    // Push children in reverse order to process them in left-to-right order
+    for (auto it = children.rbegin(); it != children.rend(); ++it)
+    {
+        nodeStack.push(*it);
+    }
```
Comment (a-zakir) [inline src/solver/expressions/iterators/pre-order.cpp]:
```suggestion
   for (auto it : children | std::views::reverse)
    {
        nodeStack.push(it);
    }
```

Comment (flomnes) [reply src/solver/expressions/iterators/pre-order.cpp]: It works but it makes the code a bit uglier (IMO)
```cpp
    for (auto it : children | std::views::reverse)
    {
        nodeStack.push(&*it);
    }
```

Comment (a-zakir) [reply src/solver/expressions/iterators/pre-order.cpp]: &* is for what?

Comment (flomnes) [reply src/solver/expressions/iterators/pre-order.cpp]: You're right, it's not necessary

**Review by a-zakir — APPROVED** [review]

**Pull request merged**